ehcache-impl/src/main/java/org/ehcache/impl/persistence/FileBasedStateRepository.java

```diff
@@ -100,8 +99,14 @@ public <K extends Serializable, V extends Serializable> StateHolder<K, V> getPer
                                                                                                      Class<K> keyClass,
                                                                                                      Class<V> valueClass,
                                                                                                      Predicate<Class<?>> isClassPermitted,
-                                                                                                     ClassLoader classLoader) {
-    // isClassPermitted and  classLoader are ignored because this state repository has already being read from file and cached in
+                                                                                                     ClassLoader classLoader) throws CachePersistenceException {
+    // Lazy loading of loadMaps using class loader in order TCCL, CacheManager CL, Custom Classloader
```

Comment:
What's the intent behind this change of moving the `loadMaps` call from constructor to this method? Why are we loading the map at every call of `getPersistentHolder`? What's the point of loading in the map, if we are calling it every time?

Reply:
To enable the Lazy loading and implement class loader ordering and passing custom class loader as parameter for de-serialization we have moved this method to getPersistentHolder(). 
getPersistentHolder() has the parameter class loader and to use this, we have moved loadMaps call from constructor to getPersistentHolder().

Reply:
There is nothing "lazy" about loading the user provided classes here. All these operations are called during the init phase of `Store.Provider` initialize the serializers. Please look up the call stack of this method and also you read more about Serializers in user doc https://www.ehcache.org/documentation/3.10/serializers-copiers.html. Effectively all the user provided classes which are Keys & Values types used in cache are loaded when the Cache is initialized.

Reply:
Ignore my previous comment, delaying loading the classes till the getPersistentHolder is the right thing to do.